### PR TITLE
feat(auth,user-profile): separate persisted user profiles from auth and add phone number support

### LINKED ISSUE #434: [NSW Backend][Auth, Profiles] User Profile Service
## Problem

The current auth flow is doing more than authentication. It is also carrying persisted user data inside request-level auth context, which creates unnecessary preload overhead on every request and makes the auth package harder to reuse in other systems.

At the same time, several services need centralized user data on demand rather than on every request. For example, a notification service may need a user's email address and phone number only when it is actually sending a message. Keeping that data tied to auth context makes the design less flexible and harder to scale across multiple services.

We need a generalized auth layer and a separate user profile service so that centralized user data can be fetched only when a service needs it.

## Proposed Solution

Introduce a dedicated User Profile Service that owns persisted user data and exposes it to other services on demand.

The auth package should remain focused on token parsing, principal mapping, and request authentication only. It should not preload stored user data into every request. Instead, services that need user-centric data should call the user profile service directly when needed.

Expected responsibilities:

- Auth should validate tokens and build runtime request context.
- User Profile Service should store and manage centralized user data such as:
  - user ID
  - email
  - phone number
  - organization or tenant identity
  - notification preferences
- Other services should fetch user data on demand from the user profile service.

This will make the architecture easier to reuse across services such as notification, workflow, and future integrations that need user-centric metadata.

## Additional Context

This feature is intended to support a more generalized architecture for NSW:

- Auth should be reusable without depending on user persistence.
- User profile data should be centralized for services that need it.
- Request handling should avoid unnecessary data loading when a service does not need user details.
- Notification and similar services should be able to request user data only when required.

### PR DESCRIPTION
**Pull request by sthanikan2000**
## Summary

This PR combines the user-profile foundation from PR #394 with the auth refactor that splits persisted user data out of the auth package. The result is a reusable auth layer that handles token parsing and request context only, while user record persistence moves into `internal/profile/user`.

It also preserves the phone number flow end-to-end so new users can carry notification-ready data from the IdP into persisted user records.

## Type of Change

- [x] New feature (non-breaking change which adds functionality)
- [x] Refactoring (no functional changes)

## Changes Made

- Split persisted user data into `internal/profile/user` and kept `internal/auth` focused on authentication and request context.
- Introduced `UserProfileService` as an optional dependency for auth, so auth can run without user persistence.
- Kept `tokenExtractor` required in `auth.Manager` and made the constructor enforce that contract.
- Added `phone_number` support through token claims, principal mapping, auth context creation, and user record persistence.
- Renamed the persisted table to `user_records` and updated the initial schema migration.
- Updated middleware and workflow routers to stop relying on preloaded user context and instead pass user metadata on demand.
- Updated IdP bootstrap resources so seeded users and app attributes can carry phone numbers.
- Refreshed auth tests and added coverage for the new manager contract and generic auth behavior.

## Testing

- [x] I have tested this change locally
- [x] I have added unit tests for new functionality
- [x] I have tested edge cases
- [x] All existing tests pass

Validation:
- `cd backend && go test ./...`

## Checklist

- [x] My code follows the project's style guidelines
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have checked that there are no merge conflicts

## Related Issues

Closes #434 

## Additional Notes

This PR intentionally merges the auth decoupling work with the user-profile foundation so the final result is a single coherent backend change instead of two partially overlapping PRs.

## Deployment Notes

Apply the backend schema migration for `user_records` and ensure the updated IdP bootstrap resources are used before seeding users.

### CODE REVIEW

File: backend/internal/auth/token_parser.go [hunk 1]
```diff
@@ -219,17 +221,18 @@ func (te *TokenExtractor) userPrincipalFromClaims(claims *tokenClaims) (*UserPri
...
 	}
-	if claims.OUHandle == nil {
-		return nil, fmt.Errorf("jwt missing ouHandle claim for user principal")
-	}
 	if claims.OUID == nil {
 		return nil, fmt.Errorf("jwt missing ouId claim for user principal")
 	}
+	if len(claims.Roles) == 0 {
+		return nil, fmt.Errorf("jwt missing roles claim for user principal")
+	}
```
[the remark this answers is not in the document]
Comment (sthanikan2000) [reply backend/internal/auth/token_parser.go]: Good catch. An Authn middleware should not reject this if `Roles` not there. AuthZ middleware will do this check and returns `StatusForbidden` if there is no access.

File: backend/internal/auth/middleware.go [hunk 2]
```diff
@@ -73,41 +69,39 @@ func Middleware(authService *AuthService, tokenExtractor *TokenExtractor) func(h
...
+						principal.UserPrincipal.UserID,
+						principal.UserPrincipal.Email,
+						func() string {
+							if principal.UserPrincipal.PhoneNumber != nil {
+								return *principal.UserPrincipal.PhoneNumber
+							}
+							return ""
+						}(),
+						principal.UserPrincipal.OUID,
+					)
```
[the remark this answers is not in the document]
Comment (sthanikan2000) [reply backend/internal/auth/middleware.go]: Addressed by adding `UserExists` function to the `UserProfileService`

File: backend/internal/profile/user/service.go [hunk 3]
```diff
@@ -0,0 +1,123 @@
...
+		Email:       email,
+		PhoneNumber: phone,
+		OUID:        ouID,
+		NSWData:     []byte(`{}`),
+	}
+
+	result := s.db.Create(record)
+	if result.Error != nil {
+		// Check for unique constraint violation (user already exists)
+		if result.Error.Error() == "ERROR: duplicate key value violates unique constraint \"user_records_pkey\"" {
```
[the remark this answers is not in the document]
Comment (sthanikan2000) [reply backend/internal/profile/user/service.go]: Upstream service which using the `UserProfileService` should check the existence of record before making call to create it. If it making create request directly and there is an duplicate record, this method will nit return any special error.

File: backend/internal/profile/user/service.go [hunk 4]
```diff
@@ -0,0 +1,144 @@
...
+	record := &Record{
+		UserID:      userID,
+		Email:       email,
+		PhoneNumber: phone,
+		OUID:        ouID,
+		NSWData:     []byte(`{}`),
+	}
+
+	result := s.db.Create(record)
+	if result.Error != nil {
```
Comment (ginaxu1) [inline backend/internal/profile/user/service.go:94]: `CreateUser` not idempotent, check again. best to use GORM's `ON CONFLICT DO NOTHING`

**Review by ginaxu1 — CHANGES_REQUESTED** [review]

File: backend/internal/auth/middleware.go [hunk 5]
```diff
@@ -73,41 +69,69 @@ func Middleware(authService *AuthService, tokenExtractor *TokenExtractor) func(h
...
-						OUHandle: principal.UserPrincipal.OUHandle,
-						OUID:     principal.UserPrincipal.OUID,
-						NSWData:  []byte(`{}`),
+						// user does not exist - attempt to create
+						create_err := userProfileService.CreateUser(
+							principal.UserPrincipal.UserID,
+							principal.UserPrincipal.Email,
+							func() string {
+								if principal.UserPrincipal.PhoneNumber != nil {
+									return *principal.UserPrincipal.PhoneNumber
```
Comment (ginaxu1) [inline backend/internal/auth/middleware.go]: seems this logic repeated twice (once in the creation step + again when building the UserContext). Better to handle the defaulting beforehand or simplify it (in buildAuthContext)
```go
phoneNumber := ""
if principal.UserPrincipal.PhoneNumber != nil {
    phoneNumber = *principal.UserPrincipal.PhoneNumber
}
```

**Review by ginaxu1 — CHANGES_REQUESTED** [review]

File: backend/internal/auth/middleware.go (hunk 5 above)
Comment (sthanikan2000) [reply backend/internal/auth/middleware.go]: Addressed

File: backend/internal/auth/middleware.go [hunk 6]
```diff
@@ -73,41 +69,36 @@ func Middleware(authService *AuthService, tokenExtractor *TokenExtractor) func(h
...
-						OUHandle: principal.UserPrincipal.OUHandle,
-						OUID:     principal.UserPrincipal.OUID,
-						NSWData:  []byte(`{}`),
+					default:
+						slog.Debug("user already exists - skipping creation", "user_id", *userID)
+						authCtx.User.ID = *userID
 					}
 				}
-				authCtx.User = userCtx
 			}
```
Comment (mushrafmim) [inline backend/internal/auth/middleware.go:85]:
```suggestion
if principal.UserPrincipal == nil || userProfileService == nil {
    return
}

user := principal.UserPrincipal

userID, err := userProfileService.GetOrCreateUser(
    user.UserID,
    user.Email,
    user.PhoneNumber,
    user.OUID,
)

if err != nil {
    slog.Error("failed to get or create user", "idp_user_id", user.UserID, "error", err)
    return
}

authCtx.User.ID = userID
```

Too many nested conditions, making the logic complex.

File: backend/internal/workflow/service/pre_consignment_service.go [hunk 7]
```diff
@@ -301,44 +301,48 @@ func (s *PreConsignmentService) GetPreConsignmentByID(ctx context.Context, preCo
...
+			userRecord = user.Record{
+				ID:   preConsignment.TraderID,
+				Data: contextJSON,
 			}
 
-			if err := tx.Create(&uc).Error; err != nil {
-				return fmt.Errorf("failed to create user context: %w", err)
+			if err := tx.Create(&userRecord).Error; err != nil {
+				return fmt.Errorf("failed to create user record: %w", err)
 			}
```
Comment (mushrafmim) [inline backend/internal/workflow/service/pre_consignment_service.go]: Why do we need to create a user record here?

File: backend/internal/workflow/service/pre_consignment_service.go [hunk 8]
```diff
@@ -348,9 +352,9 @@ func (s *PreConsignmentService) syncTraderContextToAuth(tx *gorm.DB, preConsignm
 		return fmt.Errorf("failed to marshal updated user context: %w", err)
 	}
 
-	uc.NSWData = updatedJSON
-	if err := tx.Save(&uc).Error; err != nil {
-		return fmt.Errorf("failed to update user context: %w", err)
+	// Update the user record's NSWData field
+	if err := tx.Model(&userRecord).Update("nsw_data", updatedJSON).Error; err != nil {
+		return fmt.Errorf("failed to update user NSWData: %w", err)
```
Comment (mushrafmim) [inline backend/internal/workflow/service/pre_consignment_service.go]: This should be removed, since we don't store company level data to a specific user anymore.

**Review by mushrafmim — CHANGES_REQUESTED** [review]

File: backend/internal/profile/user/service.go (hunk 4 above)
Comment (sthanikan2000) [reply backend/internal/profile/user/service.go:94]: Added a UserExists check at the top of CreateUser, returning ErrUserAlreadyExists if a record is already present

File: backend/internal/workflow/service/pre_consignment_service.go (hunk 7 above)
Comment (sthanikan2000) [reply backend/internal/workflow/service/pre_consignment_service.go]: I remove the logics. Since we don't have company profile now

File: backend/internal/auth/middleware.go (hunk 6 above)
Comment (sthanikan2000) [reply backend/internal/auth/middleware.go:85]: Addressed

**Review by ginaxu1 — DISMISSED** [review]
lgtm

File: backend/internal/auth/context.go [hunk 9]
```diff
@@ -2,20 +2,63 @@ package auth
...
+	//
+	// Implementation notes:
+	//   - Should be idempotent: calling multiple times with same idpUserID should be safe
+	//   - Called during first login after token validation
+	//   - Errors are logged but don't block authentication
+	//   - Should not return error if user already exists
+	//   - If err is non-nil, the returned user ID should be nil
+	// Returns user ID of the created or existing user, or an error if the operation fails.
+	GetOrCreateUser(idpUserID, email, phone, organizationID string) (*string, error)
 }
```
Comment (mushrafmim) [inline backend/internal/auth/context.go:49]: Shouldn't this be moved to `manager.go`?

**Review by ginaxu1 — APPROVED** [review]

**Review by mushrafmim — APPROVED** [review]
